assets/pages/regular-contributions/components/mustSignIn.jsx

```diff
@@ -0,0 +1,76 @@
+
+// @flow
+
+// ----- Imports ----- //
+
+import type { Contrib } from 'helpers/contributions';
+import React from 'react';
+import { getBaseDomain } from 'helpers/url';
+import { canContributeWithoutSigningIn, type UserTypeFromIdentityResponse } from 'helpers/identityApis';
+import AnimatedDots from 'components/spinners/animatedDots';
+import { classNameWithModifiers } from 'helpers/utilities';
+
+// ---- Types ----- //
+
+type PropTypes = {|
+  isSignedIn: boolean,
+  userTypeFromIdentityResponse: UserTypeFromIdentityResponse,
+  contributionType: Contrib,
+  checkoutFormHasBeenSubmitted: boolean,
+|};
+
+
+// ----- Functions ----- //
+
+// Build signout URL from given return URL or current location.
+function buildUrl(): string {
+
+  const encodedReturn = encodeURIComponent(window.location);
+
+  return `https://profile.${getBaseDomain()}/signin?returnUrl=${encodedReturn}`;
```

Comment:
can use the same url as https://github.com/guardian/support-frontend/pull/1116/files
I can add this after merging

Reply:
yes!